### PR TITLE
cleanup logic related with OutOfDisk

### PR DESCRIPTION
**Pull request by Huang-Wei**
/kind cleanup

**What this PR does / why we need it**:

- cleanup OOD logic in scheduling and node controller
- update comments and testcases

**Which issue(s) this PR fixes**:
- Fixes partial of https://github.com/kubernetes/kubernetes/issues/66348#issuecomment-410419351
- Some cleanup stuff is done in "update logic of adding default DS pod tolerations #67036"
- Relates "Upgrade TaintNodesByCondition to Beta #62109"
- Relates #48843

**Special notes for your reviewer**:

There are still some occurrences of `OutOfDisk`:

- in api files

```
pkg/apis/core/types.go
3591:    // NodeOutOfDisk means the kubelet will not accept new pods due to insufficient free disk
3593:    NodeOutOfDisk NodeConditionType = "OutOfDisk"

staging/src/k8s.io/api/core/v1/types.go
4027:    // NodeOutOfDisk means the kubelet will not accept new pods due to insufficient free disk
4029:    NodeOutOfDisk NodeConditionType = "OutOfDisk"
```

It's probably safer to leave it as is.

- in kubelet side

```
pkg/kubelet/kubelet_node_status.go
500:		nodestatus.OutOfDiskCondition(kl.clock.Now, kl.recordNodeStatusEvent),

pkg/kubelet/kubelet_node_status_test.go
237:							Type:               v1.NodeOutOfDisk,
363:					Type:               v1.NodeOutOfDisk,
431:					Type:               v1.NodeOutOfDisk,
644:					Type:               v1.NodeOutOfDisk,

pkg/kubelet/nodestatus/setters.go
662:// OutOfDiskCondition returns a Setter that updates the v1.NodeOutOfDisk condition on the node.
664:func OutOfDiskCondition(nowFunc func() time.Time, // typically Kubelet.clock.Now
671:		// Check if NodeOutOfDisk condition already exists and if it does, just pick it up for update.
673:			if node.Status.Conditions[i].Type == v1.NodeOutOfDisk {
683:			nodeOODCondition.Type = v1.NodeOutOfDisk

pkg/kubelet/eviction/eviction_manager.go
145:		// admit it if tolerates memory pressure taint, fail for other tolerations, e.g. OutOfDisk.
```

Will leave this for sig-node to handle.

**Release note**:
```release-note
NONE
```

/sig scheduling
/sig node

### CODE REVIEW

**Comment by Huang-Wei** [comment]
/hold

It's better (not necessary though) to wait for #67036 to be merged.

**Comment by k82cn** [comment]
please help to rebase.

**Comment by Huang-Wei** [comment]
Leave the CI errors as is. As this PR needs to built on top of #67036.

**Comment by warmchang** [comment]
Please rebase the PR.

**Comment by Huang-Wei** [comment]
Some comments in `kube_features.go` are updated. Kindly ping @bsalamat for approval.

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go [hunk 1]
```diff
@@ -976,7 +972,7 @@ func (nc *Controller) tryUpdateNodeHealth(node *v1.Node) (time.Duration, v1.Node
 
 		// remaining node conditions should also be set to Unknown
 		remainingNodeConditionTypes := []v1.NodeConditionType{
-			v1.NodeOutOfDisk,
+			v1.NodePIDPressure,
```
Comment (bsalamat) [inline pkg/controller/nodelifecycle/node_lifecycle_controller.go]: Repeated twice. It exists at line 978.

File: pkg/scheduler/core/generic_scheduler_test.go [hunk 2]
```diff
@@ -1229,7 +1229,7 @@ func TestNodesWherePreemptionMightHelp(t *testing.T) {
 				"machine1": []algorithm.PredicateFailureReason{algorithmpredicates.ErrNodeUnderDiskPressure},
 				"machine2": []algorithm.PredicateFailureReason{algorithmpredicates.ErrNodeUnderPIDPressure},
 				"machine3": []algorithm.PredicateFailureReason{algorithmpredicates.ErrNodeUnderMemoryPressure},
-				"machine4": []algorithm.PredicateFailureReason{algorithmpredicates.ErrNodeOutOfDisk},
+				"machine4": []algorithm.PredicateFailureReason{algorithmpredicates.ErrNodeUnderDiskPressure},
```
Comment (bsalamat) [inline pkg/scheduler/core/generic_scheduler_test.go]: `machine1` is already tested for disk pressure.

Comment (Huang-Wei) [reply pkg/scheduler/core/generic_scheduler_test.go]: Good catch, thanks!

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go (hunk 1 above)
Comment (Huang-Wei) [reply pkg/controller/nodelifecycle/node_lifecycle_controller.go]: Good catch, thanks!

**Comment by Huang-Wei** [comment]
@bsalamat comments have been addressed. PTAL.

**Review by resouer — APPROVED** [review]
It seems in api part, these comments are also referring to `NodeOutOfDisk`? For instance: https://github.com/kubernetes/kubernetes/blob/12be140dcd544046aba802259ee0128e17f56eb1/staging/src/k8s.io/apiserver/pkg/apis/example/types.go#L58-L59

You can find several occurrence by global search.

**Comment by Huang-Wei** [comment]
PR rebased. Kindly ping @bsalamat for /lgtm and /approve.

**Review by bsalamat — APPROVED** [review]
/lgtm
/approve

**Pull request merged**